cmake/CompileBoost.cmake

```diff
@@ -1,26 +1,29 @@
-find_package(Boost 1.72)
+if(Boost_ROOT)
```

Comment:
`BOOST_ROOT` is a magic variable to control `FindBoost.cmake` - using `Boost_ROOT` for a slightly different purpose is probably not a good idea

Reply:
I guess a better approach is to:

```
find_package(Boost 1.72)
if(NOT Boost_FOUND)
   # Download Boost, if possible
   ...
   set(Boost_ROOT place_Boost_downloaded)
   find_package(Boost 1.72)
endif()

# Forward the result of FindBoost.cmake to our own variables
set(FDB_Boost_FOUND Boost_FOUND)
...
```